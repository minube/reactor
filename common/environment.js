import { DeviceInfo, Platform } from 'react-native';
import PKG from '../package.json';

const { NODE_ENV, REACT_APP_ENV = 'development' } = process.env;
const IS_SERVER = typeof window === 'undefined';
const IS_WEB = Platform.OS === 'web';
const IS_PRODUCTION = NODE_ENV === 'production' && NODE_ENV === REACT_APP_ENV;
const IS_MOBILE_WEB = IS_WEB && !IS_SERVER
  ? /iPhone|Android|Tablet|iPad|Mobile|iPod|Windows Phone|Blackberry/i.test(DeviceInfo.userAgent)
  : false;

if (IS_WEB && !IS_SERVER) window.reactor = { version: PKG.version };

export default {
  IS_NATIVE: !IS_WEB,
  IS_SERVER: IS_SERVER && IS_WEB,
  IS_WEB,
  IS_MOBILE_WEB,
  IS_PRODUCTION,
  IS_DEVELOPMENT: !IS_PRODUCTION,
  IS_TEST: NODE_ENV === 'test',

  NODE_ENV,

  PKG: {
    name: PKG.name,
    version: PKG.version,
  },

  REACT_APP_ENV,
};
