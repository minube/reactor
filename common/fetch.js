const DEFAULT_HEADERS = {
  'Content-Type': 'application/json',
  'X-Requested-With': 'XMLHttpRequest',
};

export default async (endpoint, props = {}) => {
  const { headers, method = 'GET', ...others } = props;

  return new Promise((resolve, reject) => {
    fetch(endpoint, {
      headers: { ...DEFAULT_HEADERS, ...headers },
      method,
      ...others,
    })
      .then(async (response) => {
        const json = await response.json();

        if (response.status >= 400) reject({ code: response.status, message: json.message }); // eslint-disable-line
        else resolve(json);
      }).catch(({ message = 'ERROR_CONNECTION', response } = {}) => {
        reject({ // eslint-disable-line
          code: response ? response.status : undefined,
          message,
        });
      });
  });
};
