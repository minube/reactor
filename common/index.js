import STYLE from './style';
import THEME from './theme';

export {
  STYLE,
  THEME,
};
