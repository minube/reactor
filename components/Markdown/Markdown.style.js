import { StyleSheet } from 'react-native';

export default StyleSheet.create({

  container: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'flex-start',
    alignContent: 'flex-start',
    justifyContent: 'flex-start',
    flexWrap: 'wrap',
  },
});
