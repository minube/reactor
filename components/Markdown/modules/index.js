import markdown, { BOLD, ITALIC } from './markdown';

export {
  BOLD,
  ITALIC,
  markdown,
};
