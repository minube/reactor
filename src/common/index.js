import C from './constants';
import STYLE from './style';
import THEME from './theme';

export {
  C,
  STYLE,
  THEME,
};
