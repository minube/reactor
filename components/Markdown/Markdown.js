import { string } from 'prop-types';
import React from 'react';
import { View } from 'react-native';

import { STYLE } from '../../common';
import Text from '../Text';
import { markdown, BOLD, ITALIC } from './modules';

const Markdown = ({
  children, ...inherit
}) => {
  const sentences = markdown(children);

  return (
    sentences.length <= 1
      ?
        <Text {...inherit}>{children}</Text>
      :
        <View style={STYLE.ROW}>
          { sentences.map(({ value, type }, index) => {
            const key = `${type}${index}`;
            return (
              <Text key={key} {...inherit} bold={type === BOLD} italic={type === ITALIC}>
                {value}
              </Text>
            );
          })}
        </View>
  );
};

Markdown.propTypes = {
  children: string,
};

Markdown.defaultProps = {
  children: undefined,
};

export default Markdown;
