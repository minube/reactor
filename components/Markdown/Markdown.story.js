import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, color, boolean } from '@storybook/addon-knobs/react';

import { THEME } from '../../common';
import Markdown from './Markdown';

storiesOf('✅ Markdown', module)
  .add('default', () => (
    <Markdown>hello world</Markdown>
  ))
  .add('including *', () => (
    <Markdown>hello *world*</Markdown>
  ))
  .add('including **', () => (
    <Markdown>hello **world**.</Markdown>
  ))
  .add('combining * & **', () => (
    <Markdown>hello *world*, I'm **javi**.</Markdown>
  ))
  .add('inject a <Text> property', () => (
    <Markdown subtitle level={2}>Hello *World*</Markdown>
  ))
  .add('style container', () => (
    <Markdown styleContainer={{ width: 192, backgroundColor: THEME.COLOR.PRIMARY, padding: 16 }}>
      Lorem Ipsum is **simply dummy** text of the printing and typesetting *industry.*
    </Markdown>
  ))
  .add('🏀 Playground', () => (
    <Markdown>
      {text('content', 'Hello World')}
    </Markdown>
  ));

