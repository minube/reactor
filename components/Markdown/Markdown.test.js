import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import Markdown from './Markdown';

describe('<Markdown>', () => {
  it('renders', () => {
    const tree = renderer.create(<Markdown />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when simple text', () => {
    const tree = renderer.create(<Markdown>Hello World</Markdown>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when including *', () => {
    const tree = renderer.create(<Markdown>hello *world*</Markdown>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when including **', () => {
    const tree = renderer.create(<Markdown>hello **world**</Markdown>).toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('when combining * & **', () => {
    const tree = renderer.create(<Markdown>hello *world*, I'm **javi**.</Markdown>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('including a <Text> property', () => {
    const tree = renderer.create(<Markdown accent>Hello *World*</Markdown>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders in DOM environment', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Markdown />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

