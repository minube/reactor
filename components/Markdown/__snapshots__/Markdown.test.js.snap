// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Markdown> including a <Text> property 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "flex-start",
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
      },
      Array [],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        false,
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        Object {
          "color": "#f69a37",
        },
        Object {},
      ]
    }
  >
    Hello 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        Object {
          "fontStyle": "italic",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        Object {
          "color": "#f69a37",
        },
        Object {},
      ]
    }
  >
    World 
  </Text>
</View>
`;

exports[`<Markdown> renders 1`] = `
<View
  style={Array []}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        false,
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        Object {},
      ]
    }
  />
</View>
`;

exports[`<Markdown> when combining * & ** 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "flex-start",
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
      },
      Array [],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        false,
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        Object {},
      ]
    }
  >
    hello 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        Object {
          "fontStyle": "italic",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        Object {},
      ]
    }
  >
    world, I'm 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        Object {
          "fontWeight": "bold",
        },
        false,
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        Object {},
      ]
    }
  >
    javi. 
  </Text>
</View>
`;

exports[`<Markdown> when including * 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "flex-start",
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
      },
      Array [],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        false,
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        Object {},
      ]
    }
  >
    hello 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        Object {
          "fontStyle": "italic",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        Object {},
      ]
    }
  >
    world 
  </Text>
</View>
`;

exports[`<Markdown> when including ** 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "flex-start",
        "alignItems": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
      },
      Array [],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        false,
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        Object {},
      ]
    }
  >
    hello 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        Object {
          "fontWeight": "bold",
        },
        false,
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        Object {},
      ]
    }
  >
    world 
  </Text>
</View>
`;

exports[`<Markdown> when simple text 1`] = `
<View
  style={Array []}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        false,
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        Object {},
      ]
    }
  >
    Hello World
  </Text>
</View>
`;
